includes/MirahezeIRCRCFeedFormatter.php

```diff
@@ -77,27 +82,12 @@ public function getLine( array $feed, RecentChange $rc, $actionComment ) {
 			// IRC API which expects "Log".
 			$titleObj = Title::newFromText( 'Log/' . $attribs['rc_log_type'], NS_SPECIAL );
 		} else {
-			$titleObj = $rc->getTitle();
+			$titleObj = Title::castFromPageReference( $rc->getPage() ) ?: Title::makeTitle( NS_SPECIAL, 'BadTitle' );
 		}
 		$title = $titleObj->getPrefixedText();
 		$title = self::cleanupForIRC( $title );
 
-		if ( $attribs['rc_type'] == RC_LOG ) {
-			$url = '';
-		} else {
-			$url = $canonicalServer . $script;
-			if ( $attribs['rc_type'] == RC_NEW ) {
-				$query = '?oldid=' . $attribs['rc_this_oldid'];
-			} else {
-				$query = '?diff=' . $attribs['rc_this_oldid'] . '&oldid=' . $attribs['rc_last_oldid'];
-			}
-			if ( $useRCPatrol || ( $attribs['rc_type'] == RC_NEW && $useNPPatrol ) ) {
-				$query .= '&rcid=' . $attribs['rc_id'];
-			}
-
-			Hooks::runner()->onIRCLineURL( $url, $query, $rc );
```

Comment:
I don't think we use this anywhere, but wouldn't be a bad idea to check.

Reply:
Also, what is this way of running hooks? Does this even work nowadays?

Reply:
This hook is still executed it is just part of getNotifyUrl()

Reply:
you're right. No concerns at all from me then.